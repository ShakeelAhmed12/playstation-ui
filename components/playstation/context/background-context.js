import { createContext, useState } from 'react';

export const BackgroundContext = createContext();

export const BackgroundProvider = ({ children }) => {
    const [activeGame, setActiveGame] = useState('explore');

    return (
        <BackgroundContext.Provider value={{ activeGame, setActiveGame }}>
            {children}
        </BackgroundContext.Provider>
    )
}