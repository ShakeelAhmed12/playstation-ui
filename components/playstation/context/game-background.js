import { createContext, useState } from 'react';

export const BackgroundContext = createContext();

export const GameBackground = ({ children }) => {
    const [activeGame, setActiveGame] = useState('explore');

    return (
        <BackgroundContext.Provider value={{ activeGame, setActiveGame }}>
            {children}
        </BackgroundContext.Provider>
    )
}