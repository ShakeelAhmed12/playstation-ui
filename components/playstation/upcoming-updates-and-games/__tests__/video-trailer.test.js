import React from 'react';
import renderer from 'react-test-renderer';
import { GameBackground } from '../../context/game-background';

import VideoTrailers from '../video-trailers';

it('Video trailers renders correctly', () => {
    const tree = renderer
        .create(
            <GameBackground>
                <VideoTrailers />
            </GameBackground>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});