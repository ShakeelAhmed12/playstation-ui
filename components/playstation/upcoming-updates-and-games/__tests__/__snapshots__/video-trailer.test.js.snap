// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video trailers renders correctly 1`] = `
<section
  className="video-trailers__StyledVideoTrailers-sc-1yrdo2w-0 kvERfs"
>
  <div
    className="game-trailer-cards__StyledVideoItem-a9c8cw-0 erWEWj"
    onClick={[Function]}
  >
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "display": "block",
          "margin": 0,
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "block",
            "paddingTop": "57.14285714285714%",
          }
        }
      />
      <img
        alt="Grand Theft Auto 5"
        decoding="async"
        src="/_next/image?url=%2Fplaystation%2Fvideo-thumbnails%2Fgta-5.jpg&w=1200&q=75"
        srcSet="/_next/image?url=%2Fplaystation%2Fvideo-thumbnails%2Fgta-5.jpg&w=320&q=75 320w, /_next/image?url=%2Fplaystation%2Fvideo-thumbnails%2Fgta-5.jpg&w=420&q=75 420w, /_next/image?url=%2Fplaystation%2Fvideo-thumbnails%2Fgta-5.jpg&w=768&q=75 768w, /_next/image?url=%2Fplaystation%2Fvideo-thumbnails%2Fgta-5.jpg&w=1024&q=75 1024w, /_next/image?url=%2Fplaystation%2Fvideo-thumbnails%2Fgta-5.jpg&w=1200&q=75 1200w"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "visibility": "visible",
            "width": 0,
          }
        }
      />
    </div>
    <div
      className="video-info"
    >
      <div
        className="circle-player"
      >
        <svg
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M8 5v14l11-7z"
          />
        </svg>
      </div>
      <span
        className="title"
      >
        Grand Theft Auto 5
      </span>
    </div>
  </div>
  <div
    className="game-trailer-cards__StyledVideoItem-a9c8cw-0 erWEWj"
    onClick={[Function]}
  >
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "display": "block",
          "margin": 0,
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "block",
            "paddingTop": "57.14285714285714%",
          }
        }
      />
      <img
        alt="Spider-Man: Miles Morales"
        decoding="async"
        src="/_next/image?url=%2Fplaystation%2Fvideo-thumbnails%2Fspider-man.jpg&w=1200&q=75"
        srcSet="/_next/image?url=%2Fplaystation%2Fvideo-thumbnails%2Fspider-man.jpg&w=320&q=75 320w, /_next/image?url=%2Fplaystation%2Fvideo-thumbnails%2Fspider-man.jpg&w=420&q=75 420w, /_next/image?url=%2Fplaystation%2Fvideo-thumbnails%2Fspider-man.jpg&w=768&q=75 768w, /_next/image?url=%2Fplaystation%2Fvideo-thumbnails%2Fspider-man.jpg&w=1024&q=75 1024w, /_next/image?url=%2Fplaystation%2Fvideo-thumbnails%2Fspider-man.jpg&w=1200&q=75 1200w"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "visibility": "visible",
            "width": 0,
          }
        }
      />
    </div>
    <div
      className="video-info"
    >
      <div
        className="circle-player"
      >
        <svg
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M8 5v14l11-7z"
          />
        </svg>
      </div>
      <span
        className="title"
      >
        Spider-Man: Miles Morales
      </span>
    </div>
  </div>
  <div
    className="game-trailer-cards__StyledVideoItem-a9c8cw-0 erWEWj"
    onClick={[Function]}
  >
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "display": "block",
          "margin": 0,
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "block",
            "paddingTop": "57.14285714285714%",
          }
        }
      />
      <img
        alt="Assassin's Creed Valhalla"
        decoding="async"
        src="/_next/image?url=%2Fplaystation%2Fvideo-thumbnails%2Fassassins-creed.jpg&w=1200&q=75"
        srcSet="/_next/image?url=%2Fplaystation%2Fvideo-thumbnails%2Fassassins-creed.jpg&w=320&q=75 320w, /_next/image?url=%2Fplaystation%2Fvideo-thumbnails%2Fassassins-creed.jpg&w=420&q=75 420w, /_next/image?url=%2Fplaystation%2Fvideo-thumbnails%2Fassassins-creed.jpg&w=768&q=75 768w, /_next/image?url=%2Fplaystation%2Fvideo-thumbnails%2Fassassins-creed.jpg&w=1024&q=75 1024w, /_next/image?url=%2Fplaystation%2Fvideo-thumbnails%2Fassassins-creed.jpg&w=1200&q=75 1200w"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "visibility": "visible",
            "width": 0,
          }
        }
      />
    </div>
    <div
      className="video-info"
    >
      <div
        className="circle-player"
      >
        <svg
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M8 5v14l11-7z"
          />
        </svg>
      </div>
      <span
        className="title"
      >
        Assassin's Creed Valhalla
      </span>
    </div>
  </div>
  <div
    className="game-trailer-cards__StyledVideoItem-a9c8cw-0 erWEWj"
    onClick={[Function]}
  >
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "display": "block",
          "margin": 0,
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "block",
            "paddingTop": "57.14285714285714%",
          }
        }
      />
      <img
        alt="God Of War"
        decoding="async"
        src="/_next/image?url=%2Fplaystation%2Fvideo-thumbnails%2Fgod-of-war.jpg&w=1200&q=75"
        srcSet="/_next/image?url=%2Fplaystation%2Fvideo-thumbnails%2Fgod-of-war.jpg&w=320&q=75 320w, /_next/image?url=%2Fplaystation%2Fvideo-thumbnails%2Fgod-of-war.jpg&w=420&q=75 420w, /_next/image?url=%2Fplaystation%2Fvideo-thumbnails%2Fgod-of-war.jpg&w=768&q=75 768w, /_next/image?url=%2Fplaystation%2Fvideo-thumbnails%2Fgod-of-war.jpg&w=1024&q=75 1024w, /_next/image?url=%2Fplaystation%2Fvideo-thumbnails%2Fgod-of-war.jpg&w=1200&q=75 1200w"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "visibility": "visible",
            "width": 0,
          }
        }
      />
    </div>
    <div
      className="video-info"
    >
      <div
        className="circle-player"
      >
        <svg
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M8 5v14l11-7z"
          />
        </svg>
      </div>
      <span
        className="title"
      >
        God Of War
      </span>
    </div>
  </div>
  <div
    className="game-trailer-cards__StyledVideoItem-a9c8cw-0 erWEWj"
    onClick={[Function]}
  >
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "display": "block",
          "margin": 0,
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "block",
            "paddingTop": "57.14285714285714%",
          }
        }
      />
      <img
        alt="Call Of Duty Black Ops Cold War"
        decoding="async"
        src="/_next/image?url=%2Fplaystation%2Fvideo-thumbnails%2Fcod.jpg&w=1200&q=75"
        srcSet="/_next/image?url=%2Fplaystation%2Fvideo-thumbnails%2Fcod.jpg&w=320&q=75 320w, /_next/image?url=%2Fplaystation%2Fvideo-thumbnails%2Fcod.jpg&w=420&q=75 420w, /_next/image?url=%2Fplaystation%2Fvideo-thumbnails%2Fcod.jpg&w=768&q=75 768w, /_next/image?url=%2Fplaystation%2Fvideo-thumbnails%2Fcod.jpg&w=1024&q=75 1024w, /_next/image?url=%2Fplaystation%2Fvideo-thumbnails%2Fcod.jpg&w=1200&q=75 1200w"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "visibility": "visible",
            "width": 0,
          }
        }
      />
    </div>
    <div
      className="video-info"
    >
      <div
        className="circle-player"
      >
        <svg
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M8 5v14l11-7z"
          />
        </svg>
      </div>
      <span
        className="title"
      >
        Call Of Duty Black Ops Cold War
      </span>
    </div>
  </div>
  <div
    className="game-trailer-cards__StyledVideoItem-a9c8cw-0 erWEWj"
    onClick={[Function]}
  >
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "display": "block",
          "margin": 0,
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "block",
            "paddingTop": "57.14285714285714%",
          }
        }
      />
      <img
        alt="Godfall"
        decoding="async"
        src="/_next/image?url=%2Fplaystation%2Fvideo-thumbnails%2Fgodfall.jpg&w=1200&q=75"
        srcSet="/_next/image?url=%2Fplaystation%2Fvideo-thumbnails%2Fgodfall.jpg&w=320&q=75 320w, /_next/image?url=%2Fplaystation%2Fvideo-thumbnails%2Fgodfall.jpg&w=420&q=75 420w, /_next/image?url=%2Fplaystation%2Fvideo-thumbnails%2Fgodfall.jpg&w=768&q=75 768w, /_next/image?url=%2Fplaystation%2Fvideo-thumbnails%2Fgodfall.jpg&w=1024&q=75 1024w, /_next/image?url=%2Fplaystation%2Fvideo-thumbnails%2Fgodfall.jpg&w=1200&q=75 1200w"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "visibility": "visible",
            "width": 0,
          }
        }
      />
    </div>
    <div
      className="video-info"
    >
      <div
        className="circle-player"
      >
        <svg
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M8 5v14l11-7z"
          />
        </svg>
      </div>
      <span
        className="title"
      >
        Godfall
      </span>
    </div>
  </div>
  <div
    className="game-trailer-cards__StyledVideoItem-a9c8cw-0 erWEWj"
    onClick={[Function]}
  >
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "display": "block",
          "margin": 0,
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "block",
            "paddingTop": "57.14285714285714%",
          }
        }
      />
      <img
        alt="Cyberpunk 2077"
        decoding="async"
        src="/_next/image?url=%2Fplaystation%2Fvideo-thumbnails%2Fcyberpunk.jpg&w=1200&q=75"
        srcSet="/_next/image?url=%2Fplaystation%2Fvideo-thumbnails%2Fcyberpunk.jpg&w=320&q=75 320w, /_next/image?url=%2Fplaystation%2Fvideo-thumbnails%2Fcyberpunk.jpg&w=420&q=75 420w, /_next/image?url=%2Fplaystation%2Fvideo-thumbnails%2Fcyberpunk.jpg&w=768&q=75 768w, /_next/image?url=%2Fplaystation%2Fvideo-thumbnails%2Fcyberpunk.jpg&w=1024&q=75 1024w, /_next/image?url=%2Fplaystation%2Fvideo-thumbnails%2Fcyberpunk.jpg&w=1200&q=75 1200w"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "visibility": "visible",
            "width": 0,
          }
        }
      />
    </div>
    <div
      className="video-info"
    >
      <div
        className="circle-player"
      >
        <svg
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M8 5v14l11-7z"
          />
        </svg>
      </div>
      <span
        className="title"
      >
        Cyberpunk 2077
      </span>
    </div>
  </div>
  <div
    className="game-trailer-cards__StyledVideoItem-a9c8cw-0 erWEWj"
    onClick={[Function]}
  >
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "display": "block",
          "margin": 0,
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "block",
            "paddingTop": "57.14285714285714%",
          }
        }
      />
      <img
        alt="Far Cry 6"
        decoding="async"
        src="/_next/image?url=%2Fplaystation%2Fvideo-thumbnails%2Ffar-cry.jpg&w=1200&q=75"
        srcSet="/_next/image?url=%2Fplaystation%2Fvideo-thumbnails%2Ffar-cry.jpg&w=320&q=75 320w, /_next/image?url=%2Fplaystation%2Fvideo-thumbnails%2Ffar-cry.jpg&w=420&q=75 420w, /_next/image?url=%2Fplaystation%2Fvideo-thumbnails%2Ffar-cry.jpg&w=768&q=75 768w, /_next/image?url=%2Fplaystation%2Fvideo-thumbnails%2Ffar-cry.jpg&w=1024&q=75 1024w, /_next/image?url=%2Fplaystation%2Fvideo-thumbnails%2Ffar-cry.jpg&w=1200&q=75 1200w"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "visibility": "visible",
            "width": 0,
          }
        }
      />
    </div>
    <div
      className="video-info"
    >
      <div
        className="circle-player"
      >
        <svg
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M8 5v14l11-7z"
          />
        </svg>
      </div>
      <span
        className="title"
      >
        Far Cry 6
      </span>
    </div>
  </div>
  <div
    className="game-trailer-cards__StyledVideoItem-a9c8cw-0 erWEWj"
    onClick={[Function]}
  >
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "display": "block",
          "margin": 0,
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "block",
            "paddingTop": "57.14285714285714%",
          }
        }
      />
      <img
        alt="Watch Dogs: Legion"
        decoding="async"
        src="/_next/image?url=%2Fplaystation%2Fvideo-thumbnails%2Fwatch-dogs.jpg&w=1200&q=75"
        srcSet="/_next/image?url=%2Fplaystation%2Fvideo-thumbnails%2Fwatch-dogs.jpg&w=320&q=75 320w, /_next/image?url=%2Fplaystation%2Fvideo-thumbnails%2Fwatch-dogs.jpg&w=420&q=75 420w, /_next/image?url=%2Fplaystation%2Fvideo-thumbnails%2Fwatch-dogs.jpg&w=768&q=75 768w, /_next/image?url=%2Fplaystation%2Fvideo-thumbnails%2Fwatch-dogs.jpg&w=1024&q=75 1024w, /_next/image?url=%2Fplaystation%2Fvideo-thumbnails%2Fwatch-dogs.jpg&w=1200&q=75 1200w"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "visibility": "visible",
            "width": 0,
          }
        }
      />
    </div>
    <div
      className="video-info"
    >
      <div
        className="circle-player"
      >
        <svg
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M8 5v14l11-7z"
          />
        </svg>
      </div>
      <span
        className="title"
      >
        Watch Dogs: Legion
      </span>
    </div>
  </div>
</section>
`;
